### PR TITLE
TMEDIA-516: Upgrade arc ads

### PR DESCRIPTION
**Pull request by JackHowa**
## Description
Update arcads 6 and tests

## Jira Ticket
- [TMEDIA-516](https://arcpublishing.atlassian.net/browse/TMEDIA-516)

## Acceptance Criteria
Tests pass and arcads updated

## Test Steps

1. Checkout this branch `git checkout TMEDIA-516-updgrade-arcads`
2. `rm -rf node_modules && npm i && npm t`. Tests pass
3. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/ads-block`
4. Go to a page with ads and compare corecomponents with ads 

### Before

cc
Screen Shot 2021-10-28 at 10 01 51

### After

- no issues

Screen Shot 2021-10-28 at 10 00 17

## Dependencies or Side Effects
- `jest.mock` may not have been working as expected before

### More info on jest.mocks

- The `jest.mock` may not have been working as expected before without a second parameter. This bug report reveals that the jest.mock may not be called without the second factory argument
- If that jest.mock may not have been called in some situations, then it may not have been needed before. So the update to arcads may have been breaking this
- Since the tests don't need the `jest.mock` to pass, and don't seem to really be mocking it any way, removing this jest.mock seems to be working with the updated arcads@6
- https://github.com/facebook/jest/issues/4262#issuecomment-366374998

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

# How I updated arcads from a linked module

1. Delete any preexisting node modules `rm -rf node_modules`
2. Update the package `cd blocks/ads-block && npm i arcads@6.1.1`
3. `npm i` will update the dep of the linked package. But it won't update the nested module arcads from arcads@4  -> arcads@6.  You will notice that the package-lock with the immediate dep is not update
4. Check linked package `npm ls -g --depth=0 --link=true`. See `npm ERR! missing: arcads@^6.0.0, required by @wpmedia/ads-block@5.16.0`
5. `npm link blocks/ads-block --save` to update that link and its nested deps 
6. See the package-lock of the update

### CODE REVIEW

**Comment by ecarlisle** [comment]
@JackHowa , @rachelpost,
I think that this PR on ArcAds will ultimately solve this issue:
https://github.com/washingtonpost/ArcAds/pull/99

Themes is trying to rerun ArcAds tests because they are made available when imported into `fusion-news-theme-blocks`. I don't believe this is necessary, running unit tests of imported dependencies. I believe the imported dependencies should be trusted, or at least, not expected to have a tests guaranteed to run within other repositories with different unit testing configurations.

The ArcAds PR99 will prevent ArcAd test files from being brought into `fusion-news-theme-blocks`. I believe that will be a double win as it will both improve the size optics w/ ArcAds and prevent this test conflict paradigm.

**Comment by JackHowa** [comment]
interesting chromatic failing https://github.com/WPMedia/fusion-news-theme-blocks/runs/4035952937?check_suite_focus=true#step:6:333

but tests passing! 🎉

**Comment by JackHowa** [comment]
interesting update: 

```
ERR! ModuleNotFoundError: Module not found: Error: Can't resolve 'acorn' in '/Users/howardj/sites/fusion-news-theme-blocks/node_modules/acorn-jsx'
ERR!     at factory.create (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/webpack/lib/Compilation.js:925:10)
ERR!     at factory (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/webpack/lib/NormalModuleFactory.js:401:22)
ERR!     at resolver (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/webpack/lib/NormalModuleFactory.js:130:21)
ERR!     at asyncLib.parallel (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/webpack/lib/NormalModuleFactory.js:224:22)
ERR!     at /Users/howardj/sites/fusion-news-theme-blocks/node_modules/neo-async/async.js:2830:7
ERR!     at /Users/howardj/sites/fusion-news-theme-blocks/node_modules/neo-async/async.js:6877:13
ERR!     at normalResolver.resolve (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/webpack/lib/NormalModuleFactory.js:214:25)
ERR!     at doResolve (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/enhanced-resolve/lib/Resolver.js:213:14)
ERR!     at hook.callAsync (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/enhanced-resolve/lib/Resolver.js:285:5)
ERR!     at _fn0 (eval at create (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
ERR!     at resolver.doResolve (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/enhanced-resolve/lib/UnsafeCachePlugin.js:44:7)
ERR!     at hook.callAsync (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/enhanced-resolve/lib/Resolver.js:285:5)
ERR!     at _fn0 (eval at create (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
ERR!     at hook.callAsync (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/enhanced-resolve/lib/Resolver.js:285:5)
ERR!     at _fn0 (eval at create (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:27:1)
ERR!     at resolver.doResolve (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:67:43)
ERR!  ModuleNotFoundError: Module not found: Error: Can't resolve 'acorn' in '/Users/howardj/sites/fusion-news-theme-blocks/node_modules/acorn-jsx'
ERR!     at factory.create (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/webpack/lib/Compilation.js:925:10)
ERR!     at factory (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/webpack/lib/NormalModuleFactory.js:401:22)
ERR!     at resolver (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/webpack/lib/NormalModuleFactory.js:130:21)
ERR!     at asyncLib.parallel (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/webpack/lib/NormalModuleFactory.js:224:22)
ERR!     at /Users/howardj/sites/fusion-news-theme-blocks/node_modules/neo-async/async.js:2830:7
ERR!     at /Users/howardj/sites/fusion-news-theme-blocks/node_modules/neo-async/async.js:6877:13
ERR!     at normalResolver.resolve (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/webpack/lib/NormalModuleFactory.js:214:25)
ERR!     at doResolve (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/enhanced-resolve/lib/Resolver.js:213:14)
ERR!     at hook.callAsync (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/enhanced-resolve/lib/Resolver.js:285:5)
ERR!     at _fn0 (eval at create (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
ERR!     at resolver.doResolve (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/enhanced-resolve/lib/UnsafeCachePlugin.js:44:7)
ERR!     at hook.callAsync (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/enhanced-resolve/lib/Resolver.js:285:5)
ERR!     at _fn0 (eval at create (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
ERR!     at hook.callAsync (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/enhanced-resolve/lib/Resolver.js:285:5)
ERR!     at _fn0 (eval at create (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:27:1)
ERR!     at resolver.doResolve (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:67:43)
ERR! resolve 'acorn' in '/Users/howardj/sites/fusion-news-theme-blocks/node_modules/acorn-jsx'
ERR!   Parsed request is a module
ERR!   using description file: /Users/howardj/sites/fusion-news-theme-blocks/node_modules/acorn-jsx/package.json (relative path: .)
ERR!     Field 'browser' doesn't contain a valid alias configuration
ERR!     resolve as module
ERR!       /Users/howardj/sites/fusion-news-theme-blocks/node_modules/acorn-jsx/node_modules doesn't exist or is not a directory
ERR!       /Users/howardj/sites/fusion-news-theme-blocks/node_modules/node_modules doesn't exist or is not a directory
ERR!       /Users/howardj/sites/node_modules doesn't exist or is not a directory
ERR!       /Users/howardj/node_modules doesn't exist or is not a directory
ERR!       /Users/node_modules doesn't exist or is not a directory
ERR!       /node_modules doesn't exist or is not a directory
ERR!       looking for modules in /Users/howardj/sites/fusion-news-theme-blocks/node_modules
ERR!         using description file: /Users/howardj/sites/fusion-news-theme-blocks/package.json (relative path: ./node_modules)
ERR!           Field 'browser' doesn't contain a valid alias configuration
ERR!           using description file: /Users/howardj/sites/fusion-news-theme-blocks/package.json (relative path: ./node_modules/acorn)
ERR!             no extension
ERR!               Field 'browser' doesn't contain a valid alias configuration
ERR!               /Users/howardj/sites/fusion-news-theme-blocks/node_modules/acorn doesn't exist
ERR!             .mjs
ERR!               Field 'browser' doesn't contain a valid alias configuration
ERR!               /Users/howardj/sites/fusion-news-theme-blocks/node_modules/acorn.mjs doesn't exist
ERR!             .js
ERR!               Field 'browser' doesn't contain a valid alias configuration
ERR!               /Users/howardj/sites/fusion-news-theme-blocks/node_modules/acorn.js doesn't exist
ERR!             .jsx
ERR!               Field 'browser' doesn't contain a valid alias configuration
ERR!               /Users/howardj/sites/fusion-news-theme-blocks/node_modules/acorn.jsx doesn't exist
ERR!             .ts
ERR!               Field 'browser' doesn't contain a valid alias configuration
ERR!               /Users/howardj/sites/fusion-news-theme-blocks/node_modules/acorn.ts doesn't exist
ERR!             .tsx
ERR!               Field 'browser' doesn't contain a valid alias configuration
ERR!               /Users/howardj/sites/fusion-news-theme-blocks/node_modules/acorn.tsx doesn't exist
ERR!             .json
ERR!               Field 'browser' doesn't contain a valid alias configuration
ERR!               /Users/howardj/sites/fusion-news-theme-blocks/node_modules/acorn.json doesn't exist
ERR!             .cjs
ERR!               Field 'browser' doesn't contain a valid alias configuration
ERR!               /Users/howardj/sites/fusion-news-theme-blocks/node_modules/acorn.cjs doesn't exist
ERR!             as directory
ERR!               /Users/howardj/sites/fusion-news-theme-blocks/node_modules/acorn doesn't exist

File: package-lock.json [hunk 1]
```diff
@@ -19437,240 +19312,11 @@
           "version": "0.6.1",
           "resolved": "https://registry.npmjs.org/source-map/-/source-map-0.6.1.tgz",
           "integrity": "sha512-UjgapumWlbMhkBgzT7Ykc5YXUT46F0iKu8SGXq0bcwP5dz/h0Plj6enJqjz1Zbq2l5WaqYnrVbwWOWMyF3F47g==",
+          "dev": true,
           "optional": true
         }
       }
     },
-    "esdoc": {
```
Comment (JackHowa) [inline package-lock.json:19444]: all of this is removed deps from arcads

**Comment by JackHowa** [comment]
Updated storybook-related deps to ensure chromatic builds. working as expected locally now: 


# before

```
ERR! ModuleNotFoundError: Module not found: Error: Can't resolve 'acorn' in '/Users/howardj/sites/fusion-news-theme-blocks/node_modules/acorn-jsx'
ERR!     at factory.create (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/webpack/lib/Compilation.js:925:10)
ERR!     at factory (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/webpack/lib/NormalModuleFactory.js:401:22)
ERR!     at resolver (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/webpack/lib/NormalModuleFactory.js:130:21)
ERR!     at asyncLib.parallel (/Users/howardj/sites/fusion-news-theme-blocks/node_modules/webpack/lib/NormalModuleFactory.js:224:22)
ERR!     at /Users/howardj/sites/fusion-news-theme-blocks/node_modules/neo-async/async.js:2830:7
ERR!     at /Users/howardj/sites/fusion-news-them

```



# after

```
info => Preview built (49 s)
WARN asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
WARN This can impact web performance.
WARN Assets:

File: package.json [hunk 2]
```diff
@@ -43,11 +43,11 @@
...
-    "@storybook/addon-a11y": "^6.3.6",
-    "@storybook/addon-docs": "^6.3.6",
-    "@storybook/addon-essentials": "^6.3.8",
-    "@storybook/addon-knobs": "^6.2.9",
-    "@storybook/react": "^6.3.6",
+    "@storybook/addon-a11y": "^6.3.12",
+    "@storybook/addon-docs": "^6.3.12",
+    "@storybook/addon-essentials": "^6.3.12",
+    "@storybook/addon-knobs": "^6.3.1",
+    "@storybook/react": "^6.3.12",
```
Comment (JackHowa) [inline package.json:50]: had to update these after seemingly unrelated dependency broke storybook compilation

**Comment by ecarlisle** [comment]
# Test Results

- [x] Linting code actions have passed.
- [x] Ran the code locally based on the test instructions.
     -[] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [x] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [x] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

## Screens
Ads appeared as expected on page. 
image

ArcAds event listener is good for view port width changes:
image

Great job, @JackHowa !

**Review by ecarlisle — APPROVED** [review]

**Pull request merged**